packages/snap-store-mobx/src/Search/Stores/SearchPaginationStore.ts

```diff
@@ -37,22 +41,45 @@ export class SearchPaginationStore {
 		this.defaultPageSize = meta?.pagination?.defaultPageSize!;
 		this.totalPages = paginationData.totalPages!;
 
-		this.pageSizeOptions = [
-			{
-				label: `Show ${this.defaultPageSize}`,
-				value: this.defaultPageSize,
-			},
-			{
-				label: `Show ${this.defaultPageSize * 2}`,
-				value: this.defaultPageSize * 2,
-			},
-			{
-				label: `Show ${this.defaultPageSize * 3}`,
-				value: this.defaultPageSize * 3,
-			},
-		];
+		let optionArr: PageSizeOption[] = [];
+
+		if (paginationSettings?.pageSizeOptions) {
+			optionArr = paginationSettings?.pageSizeOptions.map(
+				(pageOption: any) =>
+					new PageSizeOption(this.services, this.pageSize, {
+						label: pageOption.label,
+						value: pageOption.value,
+					})
+			);
+		} else {
+			const defaultPageSizeOptions = [
+				{
+					label: `Show ${this.defaultPageSize}`,
+					value: this.defaultPageSize,
+				},
+				{
+					label: `Show ${this.defaultPageSize * 2}`,
+					value: this.defaultPageSize * 2,
+				},
+				{
+					label: `Show ${this.defaultPageSize * 3}`,
+					value: this.defaultPageSize * 3,
+				},
+			];
+
+			optionArr = defaultPageSizeOptions.map(
+				(pageOption: any) =>
+					new PageSizeOption(this.services, this.pageSize, {
+						label: pageOption.label,
+						value: pageOption.value,
+					})
+			);
```

Comment:
This is identical to the map on line 48 above - can we consolidate/refactor this?